comp/core/autodiscovery/listeners/snmp.go

```diff
@@ -23,6 +24,15 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/util/log"
 )
 
+const cacheKeyPrefix = "snmp"
+
+var (
+	autodiscoveryVar          = expvar.NewMap("snmpAutodiscovery")
```

Comment:
I think what we're trying to store looks like

```go
type AutodiscoveryStatus struct {
    DevicesScannedCount int
    DevicesFoundList []string
    CurrentDevice string
}
type AutodiscoveryStatusBySubnet map[string]AutodiscoveryStatus

func (m *AutodiscoveryStatusBySubnet) String() string {
	// Convert the struct to JSON for expvar to serve
	jsonData, err := json.Marshal(m)
	if err != nil {
		return "{}"
	}
	return string(jsonData)
}

var (
    autodiscoveryVar = &AutodiscoveryStatusBySubnet{}
)
func init() {
    expvar.Publish("snmpAutodiscovery", AutodiscoveryStatusBySubnet)
}
```

Then you can simply `expvar.Get("snmpAutodiscovery").(*AutodiscoveryStatusBySubnet)` to fetch it (w/ error handling)

Defining the exact type will make it a lot clearer and more error-proof by manipulating variables of the correct type

Reply:
🤔 Mmmh with the added caveat that this is not thread-safe so we'd need to manage mutex ourselves 😬  So it might not be worth it in the end